pkg/cmd/datamgr/cli/install/install.go

```diff
@@ -333,3 +341,25 @@ func (o *InstallOptions) getNumberOfNodes(f client.Factory) (int, error) {
 
 	return len(nodeList.Items), nil
 }
+
+// Check whether there is already existing data manager daemonset for old plugin image. If there is already existing daemonset,
+// delete the old one first.
+func (o *InstallOptions) CheckAndDeleteDatamgrDaemonset(f client.Factory) error {
+	fmt.Println("Check if there is existing data manager.")
+	clientset, err := f.KubeClient()
+	if err != nil {
+		errMsg := fmt.Sprint("Failed to get clientset.")
+		fmt.Println(errMsg)
+		return errors.New(errMsg)
+	}
+	err = clientset.AppsV1().DaemonSets(o.Namespace).Delete(context.TODO(), constants.DataManagerDaemonset, metav1.DeleteOptions{})
+	if err != nil && k8serrors.IsNotFound(err) {
```

Comment:
```
if k8serrors.IsNotFound(err) {
...
```

Reply:
We should check if err != nil first?

Reply:
No we don't have to. IsNotFound(err) will internally handle the case of `err != nil` while skip the case of `err == nil`.